core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -566,6 +574,13 @@ private[spark] class ExecutorAllocationManager(
     if (executorIds.contains(executorId)) {
       executorIds.remove(executorId)
       removeTimes.remove(executorId)
+      if (!removedExecutorIds.offer(executorId)) {
+        logWarning(s"The queue is full for tracking removed executors, will trim old " +
+          s"executor ids and offer again")
```

Comment:
IIUC, this will be a normal condition, right?  well, normal for an app that ever cycles through 10k executors, so probbaly dynamic allocation and a long-lived app, but this will just happen regularly once you hit that 10k.

Reply:
Yes, we can only print this warning the first time.